### DISCUSSION TITLE
Challenge 1 clarifications

### DISCUSSION THREAD

**Discussion by jabarragann** [surgical-robotics-ai/surgical_robotics_challenge#51] [Q&A] [answered]
Hi Adnan,

Challenge 1 description indicates that the estimated needle pose needs to be reported with respect to the ECM coordinate frame. Is this referring to the camera frame?

```self.ambf_cam_frame = Camera(ambf_client, "CameraFrame")```

I think you told me not to use that line of code and use the ECM coordinate frames instead, but I never did the change, sorry.

**Comment by jabarragann** [comment]
Another quick question. Is it possible to access the transformation between the camera left/right and the CameraFrame with the CRTK interface?

```self.ambf_cam = Camera(self.c, "cameraL")```

**Comment by adnanmunawar** [comment, marked as answer]
Hi Juan, yes, the reported pose should be w.r.t. the ‘CameraFrame’ kinematic object which is what the ECM class wraps. I would suggest again to use the ECM() class instead of the Camera() to make sure that there are no surprises when we run the evaluation on your submission 😊.

For your second question, technically yes, but only via the method interface and not the CRTK ROS interface. Secondly, the pose of the cameraL and cameraR is w.r.t. the CameraFrame. It may be simpler to get this from the ADF files as discussed here https://github.com/collaborative-robotics/surgical_robotics_challenge/discussions/42#discussioncomment-2564070

**Reply by jabarragann** [reply]
Great thanks Adnan!

**Comment by jabarragann** [comment]
Adnan is there an easy way of obtaining the orientations of the cameras with respect to the Camera frame?  In the ADF files, I see the position specified, but I am not exactly sure how to go from the   `look at: {x: 0.0, y: 0.0, z: -1.0}` and `up: {x: 0.0, y: 1.0, z: 0.0}` to rotation matrix.

**Reply by jabarragann** [reply]
I can query the matrix with the AMBF client, but ideally, I want my solution to only rely on the CRTK interface.

**Comment by adnanmunawar** [comment]
Hi Juan, to get a rotation matrix from 'location', 'lool at', and 'up' vectors you can follow the code/logic in the [setView](https://github.com/WPI-AIM/ambf/blob/ambf-2.0/ambf_framework/afFramework.cpp#L6748) method. Just remember the default standard of AMBF camera from [here](https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/docs/camera_conventions.md#difference-between-ambf-and-opencv-camera-conventions).
